tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/InstrumentationTests.cs

```diff
@@ -795,7 +799,18 @@ private void AssertNativeLoaderLogContainsString(string logDir, string requiredL
         private void AssertHasExpectedTelemetry(string echoLogFileName, ProcessResult processResult, string pointsJson, string injectResult, string injectResultReason, string injectResultClass)
         {
             using var s = new AssertionScope();
-            File.Exists(echoLogFileName).Should().BeTrue();
+
+            // Wait for the telemetry echo file to be written (with timeout)
+            // The telemetry forwarder may write the file asynchronously after process exit
+            var fileWaitTimeout = TimeSpan.FromSeconds(10);
+            var fileWaitStart = DateTime.UtcNow;
+            while (!File.Exists(echoLogFileName) && (DateTime.UtcNow - fileWaitStart) < fileWaitTimeout)
+            {
+                Thread.Sleep(100);
```

Comment:
Given this is going to sleep for up to 10s, should we make this an async `Task.Delay` instead? 🤔

Reply:
Sure, that would mitigate the delay. Done!